components/view_user_group_modal/view_user_group_modal_header/view_user_group_modal_header.tsx

```diff
@@ -0,0 +1,229 @@
+
+        actions.openModal({
+            modalId: ModalIdentifiers.ADD_USERS_TO_GROUP,
+            dialogType: AddUsersToGroupModal,
+            dialogProps: {
+                groupId,
+                backButtonCallback: props.backButtonAction,
+            },
+        });
+        props.onExited();
+    };
+
+    const goToEditGroupModal = () => {
+        const {actions, groupId} = props;
+
+        actions.openModal({
+            modalId: ModalIdentifiers.EDIT_GROUP_MODAL,
+            dialogType: UpdateUserGroupModal,
+            dialogProps: {
+                groupId,
+                backButtonCallback: props.backButtonAction,
+            },
+        });
+        props.onExited();
+    };
+
+    const showSubMenu = (source: string) => {
+        const {permissionToEditGroup, permissionToJoinGroup, permissionToLeaveGroup, permissionToArchiveGroup} = props;
+
+        return source.toLowerCase() !== 'ldap' &&
+            (
+                permissionToEditGroup ||
+                permissionToJoinGroup ||
+                permissionToLeaveGroup ||
+                permissionToArchiveGroup
+            );
+    };
+
+    const leaveGroup = async (groupId: string) => {
+        const {currentUserId, actions, decrementMemberCount} = props;
+
+        await actions.removeUsersFromGroup(groupId, [currentUserId]).then(() => {
+            decrementMemberCount();
+        });
+    };
+
+    const joinGroup = async (groupId: string) => {
+        const {currentUserId, actions, incrementMemberCount} = props;
+
+        await actions.addUsersToGroup(groupId, [currentUserId]).then(() => {
+            incrementMemberCount();
+        });
+    };
+
+    const archiveGroup = async (groupId: string) => {
+        const {actions} = props;
+
+        await actions.archiveGroup(groupId).then(() => {
+            props.backButtonCallback();
+            props.onExited();
+        });
+    };
+
+    const modalTitle = () => {
+        const {group} = props;
+
+        if (group) {
+            return (
+                <Modal.Title
+                    componentClass='h1'
+                    id='userGroupsModalLabel'
+                >
+                    {group.display_name}
+                </Modal.Title>
+            );
+        }
+        return (<></>);
+    };
+
+    const addPeopleButton = () => {
+        const {group, permissionToJoinGroup} = props;
+
+        if (group?.source.toLowerCase() !== 'ldap' && permissionToJoinGroup) {
+            return (
+                <button
+                    className='user-groups-create btn btn-md btn-primary'
+                    onClick={goToAddPeopleModal}
+                >
+                    <FormattedMessage
+                        id='user_groups_modal.addPeople'
+                        defaultMessage='AddPeople'
+                    />
+                </button>
+            );
+        }
+        return (<></>);
+    };
+
+    const subMenuButton = () => {
```

Comment:
Why is not this a component? Looks like a component, but is not taking advantage of it. It is just more complexity here and probably more rendering work.

This also apply to the other 2 functions that renders something similar to a component, but they are not components.